galasa-parent/dev.galasa.framework.api.cps/src/test/java/dev/galasa/framework/api/cps/internal/common/TestGalasaProperty.java

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.cps.internal.common;
+
+import org.junit.Test;
+
+import com.google.gson.Gson;
+
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+import static org.assertj.core.api.Assertions.*;
+
+import java.util.Map;
+
+public class TestGalasaProperty {
+    
+    static final Gson gson = GalasaGsonBuilder.build();
+
+    private String generateExpectedJson(String namespace, String propertyName, String propertyValue, String apiVersion){
+        return "{\n  \"apiVersion\": \""+apiVersion+"\",\n"+
+        "  \"kind\": \"GalasaProperty\",\n"+
+        "  \"metadata\": {\n"+
+        "    \"namespace\": \""+namespace+"\",\n"+
+        "    \"name\": \""+propertyName+"\"\n"+
+        "  },\n"+
+        "  \"data\": {\n"+
+        "    \"value\": \""+propertyValue+"\"\n  }\n}";
```

Comment:
This JSON is duplicated a lot - could it be re-used from somewhere? If the structure changes in the future, we'll have to change it everywhere

Reply:
I find that embedding json in test cases like this is painful to maintain.
It's not searchable due to all the escapes.
Better to build the json dynamically as json, then marshall it into a string.

Or use the code you are trying to test to build one, and check that it round-trips ?

Or use a templating library like moustache ?

If there is a way of embedding large amounts of json like this then we should do it.